core/dbt/clients/jinja.py

```diff
@@ -272,6 +275,13 @@ def pop(self, name):
             raise InternalException(f"popped {got}, expected {name}")
 
 
+def raise_error_func(func_name: str) -> Callable:
+    def raise_error(*args, **kwargs):
+        raise InternalException(f"{func_name} is not intended to be called here.")
```

Comment:
Will this include the name of the macro / node calling the unsupported function?

Reply:
Not really, but I can add it!

Reply:
Added for usage in models, for hooks, operation, test we can't really tell(they use `MacroGenerator`) but we don't get info about which function actually calls it unless we do some really aggressive tracking. So I just added a generic message indication it is in one of those places. Let me know if we want to be very specific.
I also included the full name `adapter.submit_python_job` in the message so shouldn't be hard to do a search

Reply:
Really solid UX improvement - thank you!